index.bs

```diff
@@ -1507,6 +1508,83 @@ these steps:
 
 </div>
 
+## Drag and Drop ## {#drag-and-drop}
+
+<xmp class=idl>
+partial interface DataTransferItem {
+    FileSystemHandle? getAsFileSystemHandle();
+};
+</xmp>
+
+During a <em>drag-and-drop operation</em>, dragged file and
+directory items are associated with [=file entries=] and [=directory entries=]
+respectively.
+
+<div class="note domintro">
+  : |handle| = item . {{getAsFileSystemHandle()}}
+  :: Returns a {{FileSystemFileHandle}} object if the dragged item is file and a {{FileSystemDirectoryHandle}}
+    object if the dragged item is a directory.
+</div>
+
+<div algorithm>
+
+The <dfn method for=DataTransferItem>getAsFileSystemHandle()</dfn> method steps are:
+
+1. If the {{DataTransferItem}} object is not in the <a spec=html>read/write
+    mode</a> or the <a spec=html>read-only mode</a>, then return null.
+
+1. If the <a spec=html>the drag data item kind</a> is not <em>File</em>, then return
+    null.
+
+1. Let |entry| be the [=entry=] representing the dragged file or directory.
+
+1. Set |entry|'s [=query permission steps=] to the [=native file system query permission steps=]
```

Comment:
nit: end sentences with a .

Reply:
Although with #200 merged, these permission bits aren't needed like this anymore. Sorry about the merge conflicts.